### PR TITLE
Inventory CSV and PDF export

### PR DESCRIPTION
**Pull request by lrnselfreliance**
## Summary

Adds an **Export** tab to Inventory with two client-side formats — **no new dependencies**:

- **CSV** — the raw items table (RFC-4180 quoting, UTF-8 BOM so Excel reads encoding correctly), generated entirely in the browser via the existing Blob/anchor download pattern.
- **PDF** — a printable full-inventory table **followed by a group-by summary table**, produced through the browser's print dialog ("Save as PDF"). The summary's grouping (and the quantity summed) is chosen with selectors in the Export tab.

## Architecture

The group-by aggregation previously lived inside `InventorySummary.js`'s `useMemo`. It's extracted into a new **`summarize.js`** — the single source of truth for field-role detection (`groupFieldsOf`/`quantityFieldsOf`/`defaultGroupKey`/`defaultQuantityKey`/`findCaloriesKey`/`findCountKey`) and aggregation (`summarizeInventory`/`sortSummaryRows`), reused by the Summary tab, the PDF summary, and the ration calculator. Count-aware totals + unit compaction stay in `units.js`.

- `findCaloriesKey`/`findCountKey` **moved** from `RationCalculator.js` → `summarize.js` (re-exported from RationCalculator for back-compat).
- The PDF print view is always-mounted at the route level (hidden on screen) so `window.print()` works from anywhere; the export grouping state is lifted to `InventoryRoute` and shared with both the Export panel and the print view.

## Files

- New: `inventoryExport.js`, `summarize.js`, `InventoryExportPanel.js`, `InventoryPrint.js` + tests (`inventoryExport.test.js`, `summarize.test.js`, `InventoryPrint.test.js`).
- Modified: `InventoryRoute.js` (Export tab, lifted state, always-mounted print), `InventorySummary.js` (uses shared helpers), `RationCalculator.js` (import move), `App.css` (`@media print` block).

## Notes

- CSV is items-only by design (a clean, re-importable single table); the summary lives in the PDF.
- Edge cases: no groupable fields → summary + selectors hidden; no quantity field → no Total column; empty inventory → no summary, export buttons disabled.

## Verification

- 179 frontend tests pass (new summarize/print/export tests; refactored Summary + moved Ration helpers green); build compiles clean.
- Live (Chrome): Export tab renders both cards; print DOM contains the items table (17 rows) + summary table; changing the group selector reactively updates the printed summary (Category=9 groups → Subcategory=12), with count-aware totals and calories.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by lrnselfreliance** [comment]
@greptile

File: app/src/components/inventory/inventoryExport.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+export function downloadCSV(filename, csv) {
+    const BOM = String.fromCharCode(0xFEFF);
+    const blob = new Blob([BOM + csv], {type: 'text/csv;charset=utf-8'});
+    const url = URL.createObjectURL(blob);
+    const a = document.createElement('a');
+    a.href = url;
+    a.download = filename;
+    a.click();
+    URL.revokeObjectURL(url);
+}
```
[the remark this answers is not in the document]
Comment (lrnselfreliance) [reply app/src/components/inventory/inventoryExport.js:48]: Fixed in 57afbe51 — the anchor is now appended to document.body before click() and removed after, and URL.revokeObjectURL runs on the next tick (setTimeout 0) so the blob stays alive until the browser starts fetching it.

File: app/src/components/inventory/summarize.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
+import {formatTotals, sumQuantities} from "./units";
+
+// Field-role detection and client-side aggregation for an inventory.  This is the single source of truth shared by
+// the Summary tab, the PDF export, and the ration calculator — the backend stores raw items and does no aggregation.
+
+// Fields the summary can group by (categorical), in schema order.
+export function groupFieldsOf(fields) {
+    return (fields || []).filter(f => ['text', 'select', 'location'].includes(f.type));
```
[the remark this answers is not in the document]
Comment (lrnselfreliance) [reply app/src/components/inventory/summarize.js:8]: Intentional, keeping as-is. Grouping by a text field is genuinely useful here — e.g. Subcategory (a text field) groups rice/wheat/canned nicely. The default group key is always Category (a select), so a user never lands on a unique-per-row text field by accident; choosing Name would indeed give one group per item, but that's an explicit opt-in. Excluding text would drop the useful Subcategory grouping, so the cost outweighs the benefit.
